homeassistant/config.py

```diff
@@ -663,6 +666,27 @@ async def async_process_ha_core_config(hass: HomeAssistant, config: dict) -> Non
     if CONF_UNIT_SYSTEM in config:
         hac.units = get_unit_system(config[CONF_UNIT_SYSTEM])
 
+    async def load_translations(_event: Event) -> None:
+        language = hass.config.language
+        _LOGGER.debug(f"Loading translations for language: {language}")
+        await async_load_state_translations_to_cache(hass, language)
+
+    async def load_translations_for_component(event: Event) -> None:
+        component = event.data.get("component")
+        if component is None or "." in str(component):
+            _LOGGER.debug(
+                f"Skipping loading translations for an invalid component: {component}"
+            )
+            return
```

Comment:
When does this happen?

Reply:
It happens just after HA start, `component_loaded` events are generated with following values:
`config.area_registry`, `config.auth`, `config.auth_provider_homeassistant`, `config.automation`, `config.config_entries`, `config.core`, `config.device_registry`, `config.entity_registry`, `config.script`, `config.scene`

When they are passed to translations loader the process end up in exceptions being thrown:

```
2023-07-21 15:24:15.645 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/core/homeassistant/config.py", line 681, in load_translations_for_component
    await async_load_state_translations_to_cache(hass, language, (component,))
  File "/core/homeassistant/helpers/translation.py", line 410, in async_load_state_translations_to_cache
    await async_load_translations(hass, language, "entity", integrations)
  File "/core/homeassistant/helpers/translation.py", line 356, in async_load_translations
    await cache.async_load(language, components)
  File "/core/homeassistant/helpers/translation.py", line 208, in async_load
    await self._async_load(language, components_to_load)
  File "/core/homeassistant/helpers/translation.py", line 247, in _async_load
    raise int_or_exc
homeassistant.loader.IntegrationNotFound: Integration 'area_registry' not found.
```